File: leetcode/Arrays_Hash/Find All Numbers Disappeared in an Array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int n = nums.size();

        unordered_set<int> s(begin(nums), end(nums));
        vector<int> res (nums.size() - s.size());
        for (int i = 1, j = 0; i <= n; i++){
            if (!s.count(i)) res[j++] = i;
        }
        return res;
    }
};

File: leetcode/Arrays_Hash/Find All Numbers Disappeared in an Array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        n = len(nums)
        new_ls = list(range(1, n + 1))
        st = set(nums)
        res = list()

        for i in range(len(new_ls)):
            if (new_ls[i] not in st):
                res.append(new_ls[i])
        return res
            